### ISSUE TITLE
Automatically check training and eval mode differences

### ISSUE THREAD

**Issue by dxoigmn** [IntelLabs/MART#165]
Model authors sometimes use `nn.Module.training` to change the control flow of their model. This is problematic because we often make the assumption a model in training more or less produces the same result as in eval mode. We should detect when this is not the case and warn the user!

**Comment by dxoigmn** [comment]
I will that it is also possible to change control on other things like whether groundtruth is present or not. Detecting this kind of behavior should probably be in scope too.
